sample/sample8-middlewares/UserController.ts

```diff
@@ -0,0 +1,56 @@
+import {Request} from "express";
+import {Controller} from "../../src/decorator/controllers";
+import {Get, Post, Put, Patch, Delete, Use} from "../../src/decorator/methods";
+import {Req, Next} from "../../src/decorator/params";
+
+@Controller()
+export class UserController {
+
+
+    @Use()
+    middleware( @Next() next: any) {
```

Comment:
Use decorator acts like actions one(GET, POST,..) but instead of calling express[action-type](router...) while registering action, I use express.use method, so the same rules applies, in this example the middleware will be called before **all route in all controllers**, since the calculated route is "/".
the main idea here is that the middlewares are controller agnostic, the decision to call them or not depends on the provided and calculated route.